### PR TITLE
Markdown linting with Remark

### PR DESCRIPTION
**Pull request by jshimko**
- Add local Markdown linting using [remark-lint](https://github.com/wooorm/remark-lint) (see updated README for usage details)
- create initial config
- TONS of linting across the whole repo

CI integration is in the works, but it looks like I'm going to need to [create a new engine for CodeClimate](https://docs.codeclimate.com/docs/building-a-code-climate-engine).  See my discussion in #74.  The only reason I didn't build/include the CI integration in this PR is I'd rather get this merged asap because it touches so many files.

### CODE REVIEW

File: admin/payment.md [hunk 1]
```diff
@@ -10,10 +10,10 @@ From the Dashboard, locate the payment package you want to enable. Click the plu
 
 Reaction supports the following payment packages:
 
- * [PayPal](https://github.com/reactioncommerce/reaction-paypal)
- * [Braintree](https://github.com/reactioncommerce/reaction-braintree)
- * [Stripe](https://github.com/reactioncommerce/reaction-stripe)
- * [Authorize.Net](https://github.com/taylorsmithgg/reaction-auth-net)
+- [PayPal](https://github.com/reactioncommerce/reaction-paypal)
```
Comment (aaronjudd) [inline admin/payment.md]: These payment package urls are not valid anymore

Comment (jshimko) [inline admin/payment.md]: Ok.  I wasn't really editing for content though.  Do you want me to update here or submit another PR?

File: developer/contributing.md [hunk 2]
```diff
@@ -1,13 +1,14 @@
 # Contributing
+
 At Reaction Commerce, we're dedicated to the open source community. In fact, we've designed our entire platform and business to grow from the passion and creativity that an open source community ignites. We've already attracted a small, dedicated team of open source contributors, and there's always room for more. If you'd like to join us, here's how to get started:
 
 **Step 1:**  If you haven't already, get Reaction running locally:
 
-```
+```sh
```
Comment (zenweasel) [inline developer/contributing.md]: Should we just replace this with `reaction init` so we are consistent?

File: developer/architecture/routing.md [hunk 3]
```diff
@@ -64,7 +64,7 @@ _Default permissions are not added automatically, as there will be a UI componen
 
 `Accounts.onCreateUser` adds default roles need for products, profiles, and checkout routes automatically if none are defined for the Shop.
 
-```javascript
```
Comment (aaronjudd) [inline developer/architecture/routing.md]: Some linters say `js`, other `javascript`. Who is right?

**Comment by aaronjudd** [comment]
👍  there some content fixes that need to be made, but we can do that in a new PR.

File: developer/architecture/routing.md (hunk 3 above)
Comment (jshimko) [inline developer/architecture/routing.md]: Yeah, both will work fine.  I just wanted to standardize and went with the short form for quicker typing.  Same deal with `bash` and `sh`.

Comment (aaronjudd) [inline developer/architecture/routing.md]: I had been trying to standardize us on the longer form, which is why I noticed...
